apps/src/code-studio/components/progress/StageLockDialog.jsx

```diff
@@ -0,0 +1,284 @@
+import React from 'react';
+import Radium from 'radium';
+import { connect } from 'react-redux';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
```

Comment:
tagging @poorvasingal and @tanyaparker if they want to weigh in on the design decisions.

Note - one other thing that's pretty different about my save button (in addition to the different color), is that it is an async operation and changes to "Saving..." while performing the operation. I guess we could make `Dialog` know how to handle this sort of thing too if we wanted.